src/main/python/pybuilder/pip_utils.py

```diff
@@ -21,14 +21,11 @@
 import sys
 
 from pybuilder.core import Dependency, RequirementsFile
-from pybuilder.pip_common import (Version,
-                                  SpecifierSet,
-                                  search_packages_info,
-                                  pip_working_set_init,
-                                  _pip_disallows_insecure_packages_by_default,
-                                  _pip_supports_constraints,
-                                  pip_version)
 from pybuilder.utils import execute_command, as_list
+# Plugin install_dependencies_plugin can reload pip_common and pip_utils. Do not use from ... import ...
+from pybuilder import pip_common
+Version = pip_common.Version
```

Comment:
Yes, what I meant was

> from pybuilder import pip_common
> Version = pip_common.Version

This is the exact equivalent of:
> from pybuilder.pip_common import Version

Version will have a reference to a class the first time the module is loaded/reloaded. There is no reason to implement a runtime resolution stanza in this case - it's verbose and non-effective. If the module is reloaded altogether there is no reason to do it at all.